src/README.md

```diff
@@ -303,35 +303,41 @@ Returns a structure holding ring statistics for the *link*:
 
 
 ## Packet (core.packet)
-   
-A *packet* is a data structure describing one of the network packets that
-is currently being processed. The packet is used to explicitly manage the
-life cycle of the packet. Packets are explicitly allocated and freed by
-using `packet.allocate` and `packet.free`. When a packet is received
-using `link.receive` its ownership is acquired by the calling app. The
-app must then ensure to either transfer the packet ownership to another
-app by calling `link.transmit` on the packet or free the packet using
-`packet.free`. Apps may only use packets they own, e.g. packets that have
-not been transmitted or freed. The number of allocatable packets is
-limited by the size of the underlying "freelist", e.g. a pool of unused
-packet objects from and to which packets are allocated and freed.
 
-— Function **packet.allocate**
+A *packet* is an FFI object of type `packet.packet_t` representing a network
+packet that is currently being processed. The packet is used to explicitly
+manage the life cycle of the packet. Packets are explicitly allocated and freed
+by using `packet.allocate` and `packet.free`. When a packet is received using
+`link.receive` its ownership is acquired by the calling app. The app must then
+ensure to either transfer the packet ownership to another app by calling
+`link.transmit` on the packet or free the packet using `packet.free`. Apps may
+only use packets they own, e.g. packets that have not been transmitted or
+freed. The number of allocatable packets is limited by the size of the
+underlying “freelist”, e.g. a pool of unused packet objects from and to which
+packets are allocated and freed.
 
-Returns a new empty packet. An an error is raised if there are no packets
-left on the freelist.
+— Ctype **packet.packet_t**
 
-— Function **packet.free** *packet*
+```
+struct packet {
+    uint8_t  data[packet.max_payload];
+    uint16_t length;
+};
+```
```

Comment:
I get why you are reluctant to documenting an API that may change, but documenting the exact ctype of packets is the whole point of this PR. Besides, we haven't made any promises with regards to API stability. If we document the ctype now, and change it next week, then we will at least have documented the API change.

I don't see a downside to documenting the current state, even if it might change soon. Instead, I see immediate gain by merging this PR now, and not let it be stuck waiting on future changes.